realm/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -0,0 +1,277 @@
+            new ThreadLocal<Map<RealmConfiguration, DynamicRealm>>() {
+                @Override
+                protected Map<RealmConfiguration, DynamicRealm> initialValue() {
+                    return new HashMap<RealmConfiguration, DynamicRealm>();
+                }
+            };
+
+    private static final ThreadLocal<Map<RealmConfiguration, Integer>> referenceCount =
+            new ThreadLocal<Map<RealmConfiguration,Integer>>() {
+                @Override
+                protected Map<RealmConfiguration, Integer> initialValue() {
+                    return new HashMap<RealmConfiguration, Integer>();
+                }
+            };
+
+    private DynamicRealm(RealmConfiguration configuration, boolean autoRefresh) {
+        super(configuration, autoRefresh);
+    }
+
+    /**
+     * Realm static constructor that returns a dynamic variant of the Realm instance defined by provided
+     * {@link io.realm.RealmConfiguration}. Dynamic Realms do not care about schemaVersion and schemas, so opening a
+     * DynamicRealm will never trigger a migration
+     *
+     * @return The DynamicRealm defined by the configuration.
+     * @see RealmConfiguration for details on how to configure a Realm.
+     */
+    public static DynamicRealm getInstance(RealmConfiguration configuration) {
+        if (configuration == null) {
+            throw new IllegalArgumentException("A non-null RealmConfiguration must be provided");
+        }
+        return create(configuration);
+    }
+
+    /**
+     * Instantiates and adds a new object to the Realm.
+     *
+     * @param className The class of the object to create.
+     * @return The new object.
+     * @throws RealmException if the object could not be created.
+     */
+    public DynamicRealmObject createObject(String className) {
+        Table table = getTable(className);
+        long rowIndex = table.addEmptyRow();
+        return get(DynamicRealmObject.class, className, rowIndex);
+    }
+
+    /**
+     * Returns a RealmQuery, which can be used to query for the provided class.
+     *
+     * @param className The class of the object which is to be queried for.
+     * @return A RealmQuery, which can be used to query for specific objects of provided type.
+     * @see io.realm.RealmQuery
+     * @throws IllegalArgumentException if the class doesn't exist.
+     */
+    public RealmQuery<DynamicRealmObject> where(String className) {
+        checkIfValid();
+        if (!sharedGroupManager.hasTable(Table.TABLE_PREFIX + className)) {
+            throw new IllegalArgumentException("Class does not exist in the Realm so it cannot be queried: " + className);
+        }
+        return RealmQuery.createDynamicQuery(this, className);
+    }
+
+    /**
+     * Removes all objects of the specified class.
+     *
+     * @param className The class for which all objects should be removed.
+     */
+    public void clear(String className) {
+        getTable(className).clear();
```

Comment:
shouldn't we `checkIfValid();` or another Thread can delete all. same for `Realm.checkIfValid();`